src/ngraph/op/batch_norm.cpp

```diff
@@ -247,16 +325,23 @@ void ngraph::op::BatchNorm::generate_adjoints(autodiff::Adjoints& adjoints,
     //and get_n() is used to sort the inputs in the same order as Batchnorm's outputs
     //Next, Mean and Variance (`at(1)` and `at(2)`) are extracted
     //Please see `add_output` in `BatchNorm::BatchNorm` for more details
-    std::vector<std::shared_ptr<Node>> goes(get_outputs().size());
 
-    for (auto _input : get_output_inputs(0))
+    std::vector<std::shared_ptr<Node>> goes(get_outputs().size());
+    if (this->get_training_flag() && get_input_size() == 3)
```

Comment:
I think we are using this pattern to order outputs in multiple spots. We should probably eventually refactor it into a helper.